src/celengine/pointstarrenderer.h

```diff
@@ -0,0 +1,57 @@
+// pointstarrenderer.h
+//
+// Copyright (C) 2001-2019, the Celestia Development Team
+// Original version by Chris Laurel <claurel@gmail.com>
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+
+#pragma once
+
+#include <Eigen/Core>
+#include <vector>
+#include "objectrenderer.h"
+#include "renderlistentry.h"
+
+class ColorTemperatureTable;
+class PointStarVertexBuffer;
+class Star;
+class StarDatabase;
+
+class PointStarRenderer : public ObjectRenderer<Star, float>
+{
+ public:
+    static constexpr const float StarDistanceLimit = 1.0e6f;
```

Comment:
this actually results in undefined reference while linking when using c++11/14, c++17 is fine

Reply:
> this actually results in undefined reference while linking when using c++11/14, c++17 is fine

omg. why all that compilers are so different? g++ eats this fine and even clang (I assume you have clang) from  xcode11.2 does.

Reply:
this happens while building on Xcode with a Xcode project rather than the cmake generated, I guess there are difference in cxx flag, but that's what I found

https://en.cppreference.com/w/cpp/language/static

If a const non-inline (since C++17) static data member or a constexpr static data member (since C++11) is odr-used, a definition at namespace scope is still required, but it cannot have an initializer. This definition is deprecated for constexpr data members (since C++17).